src/client_side_reply.cc

```diff
@@ -318,7 +324,8 @@ clientReplyContext::processExpired()
                                  http->log_uri, http->request->flags, http->request->method);
         /* NOTE, don't call StoreEntry->lock(), storeCreateEntry() does it */
 
-        if (collapsingAllowed && Store::Root().allowCollapsing(entry, http->request->flags, http->request->method)) {
+        if (collapsingAllowed && mayInitiateCollapsing() &&
+            Store::Root().allowCollapsing(entry, http->request->flags, http->request->method)) {
```

Comment:
Do you think that the construction "if collapsing is allowed then allow collapsing" is a bit misleading:
```
if (collapsingAllowed)
    allowCollapsing()
```

Reply:
Yes, the name `collapsingAllowed` is misleading. We may find a better name, but that should be done in a separate PR unless we have to change most of those collapsingAllowed lines for other reasons. My current best replacement candidate is `onCollapsingPath`, but I hope we can do better.